esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -530,7 +646,29 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
         }
         else
         {
-            conn->close(false);
+            conn->close();
+            return conn->getRoot();
+        }
+    }
+
+    virtual IPropertyTree* getBindingTree(const char* bindingId, StringBuffer& msg) override
+    {
+        if (!bindingId || !*bindingId)
+        {
+            msg.set("Could not get configuration: Target binding id not provided");
+            return nullptr;
+        }
+
+        VStringBuffer xpath("%s[@id='%s']", ESDL_BINDING_PATH, bindingId);
+        Owned<IRemoteConnection> conn = querySDS().connect(xpath.str(), myProcessSession(), RTM_LOCK_READ , SDS_LOCK_TIMEOUT_DESDL);
+        if (!conn)
+        {
+            msg.setf("Could not find binding for %s", bindingId);
+            return nullptr;
+        }
+        else
+        {
+            conn->close();
```

Comment:
I want to check with @jakesmith if this is legal. (It occurs quite a lot in this file).  Especially since close() has this comment:
    virtual void close(bool deleteRoot=false) = 0; // invalidates root.

Reply:
It is not safe, should be changed.
If you're going to close the connection, you need to finish with it's contents 1st.

Reply:
@jakesmith is the createPTreeFromIPT correct - (I think it is but want to double check).  Again I don't think the close gains anything.